### DISCUSSION TITLE
Allow relations to non primary key fields

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#8720] [Feature Requests]
Is it possibile, in Directus, to configure a relation from `CollectionA.field` to `CollectionB` where `CollectionA.field` is not an integer and is not a foreign key to `CollectionB`'s primary field?

For instance `CollectionB`'s primary key is `id`, but it also has a string field `code`. I'd like to configure Directus so that `CollectionA.field` refers to `CollectionB.code`. In MySql, of course, it is possibile to create such a foreign key. How to configure Directus to follow it?
Thanks

**Comment by rijkvanzanten** [comment]
Directus doesn't currently support foreign keys to non-primary key fields. We'll leave this discussion up as the feature request for that functionality 👍🏻

**Reply by thoeby** [reply]
Why is that? Add at least some validation upon import/configuration of new tables. Or add a note to the documentation.

**Reply by rijkvanzanten** [reply]
> Why is that?

Simply because it hasn't been implemented yet 🤷🏻 

> Add at least some validation upon import/configuration of new tables.

Did you run into any particular problem?

**Reply by thoeby** [reply]
Yes, I import spreadsheets (need to be imported regularly) and the relation between the tables is made with fields not named 'id'. Currently I have to add another table (or field) where I keep track of all id's and the corresponding field. Since those fields unique per design already, (they use the field as PK themself) there is no point in having another id field in my case.

**Reply by rijkvanzanten** [reply]
The name `id` is not required though. You can make a collection with a differently named primary key, in which case you should be able to import your spreadsheet as usual 👍🏻

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
